src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb

```diff
@@ -65,23 +67,23 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
             Return TypeOf item IsNot AbstractGenerateCodeItem
         End Function
 
-        Private Function GetTypesInFile(semanticModel As SemanticModel, cancellationToken As CancellationToken) As IEnumerable(Of INamedTypeSymbol)
+        Private Function GetTypesAndPositionsInFile(semanticModel As SemanticModel, cancellationToken As CancellationToken) As IEnumerable(Of Tuple(Of INamedTypeSymbol, Integer))
             Try
-                Dim types As New HashSet(Of INamedTypeSymbol)
+                Dim types As New Dictionary(Of INamedTypeSymbol, Integer)
                 Dim nodesToVisit As New Stack(Of SyntaxNode)
 
                 nodesToVisit.Push(DirectCast(semanticModel.SyntaxTree.GetRoot(cancellationToken), SyntaxNode))
 
                 Do Until nodesToVisit.IsEmpty
                     If cancellationToken.IsCancellationRequested Then
-                        Return SpecializedCollections.EmptyEnumerable(Of INamedTypeSymbol)()
+                        Return SpecializedCollections.EmptyEnumerable(Of Tuple(Of INamedTypeSymbol, Integer))()
                     End If
 
                     Dim node = nodesToVisit.Pop()
                     Dim type = TryCast(semanticModel.GetDeclaredSymbol(node, cancellationToken), INamedTypeSymbol)
 
                     If type IsNot Nothing Then
-                        types.Add(type)
+                        types(type) = node.SpanStart
```

Comment:
Dumb question -- is span start the right thing? That doesn't cause weird stuff in say nested class scenarios where it's viewed as the "outside" of the class vs. the "inside" of the class?